// @azure/communication-email/dist/commonjs/generated/src/models/mappers.js
"use strict";
/*
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailSendExceptionHeaders = exports.EmailSendHeaders = exports.EmailGetSendResultExceptionHeaders = exports.EmailGetSendResultHeaders = exports.EmailAttachment = exports.EmailAddress = exports.EmailRecipients = exports.EmailContent = exports.EmailMessage = exports.ErrorResponse = exports.ErrorAdditionalInfo = exports.ErrorDetail = exports.EmailSendResult = void 0;
exports.EmailSendResult = {
    type: {
        name: "Composite",
        className: "EmailSendResult",
        modelProperties: {
            id: {
                serializedName: "id",
                required: true,
                type: {
                    name: "String",
                },
            },
            status: {
                serializedName: "status",
                required: true,
                type: {
                    name: "String",
                },
            },
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorDetail",
                },
            },
        },
    },
};
exports.ErrorDetail = {
    type: {
        name: "Composite",
        className: "ErrorDetail",
        modelProperties: {
            code: {
                serializedName: "code",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            message: {
                serializedName: "message",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            target: {
                serializedName: "target",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            details: {
                serializedName: "details",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorDetail",
                        },
                    },
                },
            },
            additionalInfo: {
                serializedName: "additionalInfo",
                readOnly: true,
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "ErrorAdditionalInfo",
                        },
                    },
                },
            },
        },
    },
};
exports.ErrorAdditionalInfo = {
    type: {
        name: "Composite",
        className: "ErrorAdditionalInfo",
        modelProperties: {
            type: {
                serializedName: "type",
                readOnly: true,
                type: {
                    name: "String",
                },
            },
            info: {
                serializedName: "info",
                readOnly: true,
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } },
                },
            },
        },
    },
};
exports.ErrorResponse = {
    type: {
        name: "Composite",
        className: "ErrorResponse",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorDetail",
                },
            },
        },
    },
};
exports.EmailMessage = {
    type: {
        name: "Composite",
        className: "EmailMessage",
        modelProperties: {
            headers: {
                serializedName: "headers",
                type: {
                    name: "Dictionary",
                    value: { type: { name: "String" } },
                },
            },
            senderAddress: {
                serializedName: "senderAddress",
                required: true,
                type: {
                    name: "String",
                },
            },
            content: {
                serializedName: "content",
                type: {
                    name: "Composite",
                    className: "EmailContent",
                },
            },
            recipients: {
                serializedName: "recipients",
                type: {
                    name: "Composite",
                    className: "EmailRecipients",
                },
            },
            attachments: {
                serializedName: "attachments",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EmailAttachment",
                        },
                    },
                },
            },
            replyTo: {
                serializedName: "replyTo",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EmailAddress",
                        },
                    },
                },
            },
            disableUserEngagementTracking: {
                serializedName: "userEngagementTrackingDisabled",
                type: {
                    name: "Boolean",
                },
            },
        },
    },
};
exports.EmailContent = {
    type: {
        name: "Composite",
        className: "EmailContent",
        modelProperties: {
            subject: {
                serializedName: "subject",
                required: true,
                type: {
                    name: "String",
                },
            },
            plainText: {
                serializedName: "plainText",
                type: {
                    name: "String",
                },
            },
            html: {
                serializedName: "html",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.EmailRecipients = {
    type: {
        name: "Composite",
        className: "EmailRecipients",
        modelProperties: {
            to: {
                serializedName: "to",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EmailAddress",
                        },
                    },
                },
            },
            cc: {
                serializedName: "cc",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EmailAddress",
                        },
                    },
                },
            },
            bcc: {
                serializedName: "bcc",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "EmailAddress",
                        },
                    },
                },
            },
        },
    },
};
exports.EmailAddress = {
    type: {
        name: "Composite",
        className: "EmailAddress",
        modelProperties: {
            address: {
                serializedName: "address",
                required: true,
                type: {
                    name: "String",
                },
            },
            displayName: {
                serializedName: "displayName",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.EmailAttachment = {
    type: {
        name: "Composite",
        className: "EmailAttachment",
        modelProperties: {
            name: {
                serializedName: "name",
                required: true,
                type: {
                    name: "String",
                },
            },
            contentType: {
                serializedName: "contentType",
                required: true,
                type: {
                    name: "String",
                },
            },
            contentInBase64: {
                serializedName: "contentInBase64",
                required: true,
                type: {
                    name: "String",
                },
            },
            contentId: {
                serializedName: "contentId",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.EmailGetSendResultHeaders = {
    type: {
        name: "Composite",
        className: "EmailGetSendResultHeaders",
        modelProperties: {
            retryAfter: {
                serializedName: "retry-after",
                type: {
                    name: "Number",
                },
            },
        },
    },
};
exports.EmailGetSendResultExceptionHeaders = {
    type: {
        name: "Composite",
        className: "EmailGetSendResultExceptionHeaders",
        modelProperties: {
            xMsErrorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
exports.EmailSendHeaders = {
    type: {
        name: "Composite",
        className: "EmailSendHeaders",
        modelProperties: {
            operationLocation: {
                serializedName: "operation-location",
                type: {
                    name: "String",
                },
            },
            retryAfter: {
                serializedName: "retry-after",
                type: {
                    name: "Number",
                },
            },
        },
    },
};
exports.EmailSendExceptionHeaders = {
    type: {
        name: "Composite",
        className: "EmailSendExceptionHeaders",
        modelProperties: {
            xMsErrorCode: {
                serializedName: "x-ms-error-code",
                type: {
                    name: "String",
                },
            },
        },
    },
};
